source/deployment/cluster.rst

```diff
@@ -134,75 +134,143 @@ A sample configuration for NGINX is provided below.  It assumes you have two Mat
+````````````````````
+
+The database can be configured for high availability and transparent failover use the existing database technologies. We recommend MySQL Clustering, Postgres Clustering, or Amazon Aoura. Database transparent failover is beyond the scope of this documentation.
 
------
+Update Guide
+------------
+
+Updating configuration without service interruption
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A service interruption is not required for most configuration updates. See `Updating server with service interruption`_ for a list of configuration updates that require a service interruption.
+
+You can apply updates during a period of low load, but if your HA cluster is sized correctly, you can do it at any time. The system downtime is brief, and depends on the number of Mattermost servers in your cluster. Note that you are not restarting the machines, only the Mattermost server applications. A Mattermost server restart generally takes about 5 seconds.
+
+1. Make a backup of your existing ``config.json`` file.
+2. For one of the Mattermost servers, make the configuration changes to ``config.json`` and save the file. Do not reload the file yet.
+3. Copy the ``config.json`` file to the other servers.
+4. Shut down Mattermost on all but one server.
+5. Reload the configuration file on the server that is still running. Go to **System Console > Configuration** and click **Reload Configuration from Disk**
+6. Start the other servers. 
+
+Updating server without service interruption
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A service interruption is not required for security patch dot releases of the Mattermost server.
+
+You can apply updates during a period of low load, but if your HA cluster is sized correctly, you can do it at any time. The system downtime is brief, and depends on the number of Mattermost servers in your cluster. Note that you are not restarting the machines, only the Mattermost server applications. A Mattermost server restart generally takes about 5 seconds.
+
+1. Review the upgrade procedure in the *Upgrade Enterprise Edition* section of :doc:`../administration/upgrade`.
+2. Make a backup of your existing ``config.json`` file.
+3. Set your proxy to move all new requests to a single server. If you are using NGINX and it's configured with an upstream backend section in ``/etc/nginx/sites-available/mattermost`` then comment out all but the one server that you intend to update first, and reload NGINX.
+4. Shut down Mattermost on each server except the one that you are updating first.
+5. Upgrade each Mattermost instance that is shut down.
+6. On each server, replace the new ``config.json`` file with your backed-up copy.
+7. Start the Mattermost servers.
+8. Repeat the upgrade procedure for the server that was left running.
+
+Updating server with service interruption
```

Comment:
Propose renaming this title and two above it with something more focused on "continuous operation", which should be the theme of this document 

Proposed titles: 

- Updating configuration changes while operating continuously
- Updating server version while operating continuously
- Updating system with configuration change requiring scheduled maintenance downtime

@JeffSchering also, propose we add to Style Guide: 

1. an objective of accurately framing documentation in the context of what is to be achieved, 
2. examples of terminology in the context of achievement 

Examples of terminology in the context of achievement: 

- Use "with continuous operation" to be clear on the desired state and avoid "without service interruption" as a double negative definition.  
- Use "with scheduled maintenance downtime" instead of "with service interruption", to make it clear the system should run under IT best practices. 

We should also be clear terminology in the context of achievement should always be accurate and should avoid unintentionally overselling a feature, 

Please see if we can use the [example above on what HA offers](https://github.com/mattermost/docs/pull/825/files#r98596393) as hyperbolic illustration of a correction?